.github/workflows/lint.yml

```diff
@@ -0,0 +1,17 @@
+name: Linting
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
```

Comment:
Can these be empty?

Reply:
Yes; these appear to be working as expected and I think I found similar things as examples in the GH action docs.  I may not be remembering correctly, but I think I got a syntax error when I tried to omit the `:`

Reply:
Yes, I used that for the release workflow as well:

https://github.com/geopandas/pyogrio/blob/7fa0ded85f34f8821c8a57e6aee3faa8df133aa3/.github/workflows/release.yml#L16

This workflow_dispatch ensures you can trigger the action manually from github (without creating a new PR or so), which is useful for eg ensuring the release workflow is passing